### DISCUSSION TITLE
Perform ANN with some filter, how to get the angular similarity score?

### DISCUSSION THREAD

**Discussion by P3ngLiu** [alexklibisz/elastiknn#361] [Q&A] [answered]
@alexklibisz  I'm trying to do an ANN search query with filters. I used the function score query in your documents, but it didn't return the angular score.  I also noticed the query rescorer, but looks like it will be limited by "window_size". Can you give me some suggestions about how to do filter + ANN and get Angular similarity socre?

**Comment by alexklibisz** [comment, marked as answer]
Hi, there are examples here:
- https://elastiknn.com/tutorials/multimodal-search-amazon-products-dataset/#Combine-keyword-and-nearest-neighbors-queries-using-Elastiknn's-exact-query
- https://elastiknn.com/tutorials/multimodal-search-amazon-products-dataset/#Combine-keyword-and-nearest-neighbors-queries-using-Elastiknn's-approximate-query

If you have more questions along these lines, open up a discussion: https://github.com/alexklibisz/elastiknn/discussions
